### PR TITLE
Rename some variables that include master instead of control plane

### PR DESCRIPTION
**Pull request by gooney152**
#### What type of PR is this?
/kind cleanup
/wg naming
/sig testing
/area test

#### What this PR does / why we need it:
Refactor variables names that contains "master' to "controlplane".

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by gooney152** [comment]
references issue #[94901](https://github.com/kubernetes/kubernetes/issues/94901)

**Comment by gooney152** [comment]
I just finished signing the CLA agreement and got the conformation email sent to me.  Will this PR take a while to update? @oomichi

**Comment by MorrisLaw** [comment]
ok-to-test

**Comment by oomichi** [comment]
The change itself seems good for me.
Please sign CLA by following https://identity.linuxfoundation.org/projects/cncf

**Comment by gooney152** [comment]
@oomichi CLA signed

**Comment by gooney152** [comment]
@oomichi  thanks

**Comment by gooney152** [comment]
@MorrisLaw

File: test/e2e/framework/resource_usage_gatherer.go [hunk 1]
```diff
@@ -365,10 +365,10 @@ type NodesSet int
...
 	// AllNodes means all containers on all nodes.
 	AllNodes NodesSet = 0
-	// MasterNodes means all containers on Master nodes only.
-	MasterNodes NodesSet = 1
-	// MasterAndDNSNodes means all containers on Master nodes and DNS containers on other nodes.
-	MasterAndDNSNodes NodesSet = 2
+	// ControlPlaneNodes means all containers on ControlPlane nodes only.
+	ControlPlaneNodes NodesSet = 1
+	// ControlPlaneAndDNSNodes means all containers on ControlPlane nodes and DNS containers on other nodes.
+	ControlPlaneAndDNSNodes NodesSet = 2
```
Comment (sftim) [inline test/e2e/framework/resource_usage_gatherer.go:371]:
```suggestion
	// ControlPlaneNodes means all containers on control plane nodes only.
	ControlPlaneNodes NodesSet = 1
	// ControlPlaneAndDNSNodes means all containers on control plane nodes and DNS containers on other nodes.
	ControlPlaneAndDNSNodes NodesSet = 2
```

**Review by sftim — COMMENTED** [review]
one nit

File: test/e2e/framework/framework.go [hunk 2]
```diff
@@ -261,10 +261,10 @@ func (f *Framework) BeforeEach() {
 		var err error
 		var nodeMode NodesSet
 		switch TestContext.GatherKubeSystemResourceUsageData {
-		case "master":
-			nodeMode = MasterNodes
-		case "masteranddns":
-			nodeMode = MasterAndDNSNodes
+		case "controlplane":
```
Comment (pohly) [inline test/e2e/framework/framework.go:264]: Isn't this a functional and/or API change?

Whoever sets TestContext.GatherKubeSystemResourceUsageData must change their invocation. This needs to be announced in the changelog.

I prefer to accept "master" as an alias for "controlplane" until everyone had a chance to adapt.

File: test/e2e/framework/framework.go [hunk 3]
```diff
@@ -261,10 +261,10 @@ func (f *Framework) BeforeEach() {
...
+			nodeMode = ControlPlaneNodes
+		case "controlplaneanddns":
```
Comment (pohly) [inline test/e2e/framework/framework.go:266]: Same here. Better accept "masteranddns" as alias.

**Review by pohly — CHANGES_REQUESTED** [review]

File: test/e2e/framework/framework.go (hunk 3 above)
Comment (gooney152) [reply test/e2e/framework/framework.go:266]: @pohly  okay thanks for the feedback

**Comment by gooney152** [comment]
/assign @pohly  @sftim 
reverted back to master in those minor corrections

File: test/e2e/framework/framework.go [hunk 4]
```diff
@@ -262,9 +262,9 @@ func (f *Framework) BeforeEach() {
 		var nodeMode NodesSet
 		switch TestContext.GatherKubeSystemResourceUsageData {
 		case "master":
-			nodeMode = MasterNodes
+			nodeMode = ControlPlaneNodes
```
Comment (pohly) [inline test/e2e/framework/framework.go:265]: I think it is desirable to deprecate "master" and "masteranddns". Just do it gradually.

That means using `case "master", "controlplane"` here and add a comment that "master" can be removed i Kubernetes 1.23. Add a changelog entry in your PR description.

Also, update:
* https://github.com/kubernetes/kubernetes/blob/ebcb4a2d88c83096e6068aa56e9a5281976e1fec/test/e2e/framework/test_context.go#L130-L133
* https://github.com/kubernetes/kubernetes/blob/ebcb4a2d88c83096e6068aa56e9a5281976e1fec/test/e2e/framework/test_context.go#L290

Finally, please squash your commits.

**Pull request closed without merging**